docs/source/guides/contributor/chapters/runners.rst

```diff
@@ -568,3 +568,96 @@ for runnables with kind ``foo``.
 .. literalinclude:: ../../../../../examples/nrunner/runners/avocado-runner-foo
    :language: python
    :linenos:
+
+
+Runners messages
+----------------
+
+Every runner have to send information about the run to the avocado
+core. Those informations are sent by messages which have different
+types based on the information which they are transmitting. Avocado
+understands three main types of messages, the start, running, and
+finish. The start and finish messages are obligatory and every
+runner have to send those. The running messages can transmitting
+different information during runner run-time like logs, warnings,
+errors .etc and those informations will be processed by avocado core.
+
+Supported message types
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Start message
++++++++++++++
+This message has to be sent when the runner starts the test.
+
+:param status: 'started'
+:param time: start time of the test
+:type time: float
+:example: {'status': 'started', 'time': 16444.819830573}
+
+Finished message
+++++++++++++++++
+This message has to be sent when the runner finishes the test.
+
+:param status: 'finished'
+:param result: test result
+:type result: `avocado.core.teststatus.user_facing_status`
```

Comment:
Ideally, this would be something like:

```
Lowercase values for the statuses defined in :data:`avocado.core.teststatus.user_facing_status`
```

But I noticed sphinx wasn't happy about linking to a "data" with that name (or maybe it was because it had no documentation).  I also noticed there were obsolete statuses there.  So I came up with https://github.com/avocado-framework/avocado/pull/4510 .  If it gets reviewed/merged, it should be possible to do:

```
Lowercase values for the statuses defined in :data:`avocado.core.teststatus.STATUSES`
```